ui/app/store/actions.js

```diff
@@ -422,17 +422,18 @@ export function connectHardware(deviceName, page, hdPath) {
   };
 }
 
-export function unlockHardwareWalletAccount(index, deviceName, hdPath) {
+export function unlockHardwareWalletAccounts(indexes, deviceName, hdPath) {
   log.debug(
     `background.unlockHardwareWalletAccount`,
-    index,
+    indexes,
     deviceName,
     hdPath,
   );
-  return (dispatch) => {
+  return async (dispatch) => {
     dispatch(showLoadingIndication());
-    return new Promise((resolve, reject) => {
-      background.unlockHardwareWalletAccount(
+
+    for (const index of indexes) {
+      await background.unlockHardwareWalletAccount(
```

Comment:
`background.unlockHardwareWalletAccount` doesn't return a Promise - it uses a callback. You'll want to switch to using `promisifiedBackground` instead

Reply:
Bingo!